packages/contracts-bedrock/src/L2/RevenueSharer.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import { L2StandardBridge } from "src/L2/L2StandardBridge.sol";
+import { FeeVault } from "src/universal/FeeVault.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { SafeCall } from "src/libraries/SafeCall.sol";
+
+/// @custom:proxied
+/// @custom:predeploy 0x4200000000000000000000000000000000000024
+/// @title RevenueSharer
+/// @dev Withdraws funds from system FeeVault contracts,
+/// pays a share of revenue to a designated Beneficiary
+/// and sends the remainder to a configurable adddress on L1.
+contract RevenueSharer {
+    /*//////////////////////////////////////////////////////////////
+                            Constants
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @dev The basis point scale which revenue share splits are denominated in.
+     */
+    uint32 public constant BASIS_POINT_SCALE = 10_000;
+    /**
+     * @dev The minimum gas limit for the FeeDisburser withdrawal transaction to L1.
+     */
+    uint32 public constant WITHDRAWAL_MIN_GAS = 35_000;
+    /**
+     * @dev The percentage coeffieicnt of revenue denominated in basis points that is used in
+     *      Optimism revenue share calculation.
+     */
+    uint256 public constant REVENUE_COEFFICIENT_BASIS_POINTS = 1_500;
+    /**
+     * @dev The percentage coefficient of profit denominated in basis points that is used in
+     *      Optimism revenue share calculation.
+     */
+    uint256 public constant PROFIT_COEFFICIENT_BASIS_POINTS = 250;
+
+    /*//////////////////////////////////////////////////////////////
+                            Immutables
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @dev The address of the Optimism wallet that will receive Optimism's revenue share.
+     */
+    address payable public immutable BENEFICIARY;
```

Comment:
Can you move these into storage from being immutable + add initializer? We need an "MCP L2" project and this reduces the amount of work for that project. ie every L2 chain should be able to start with the same genesis.json then we can "preprocess" that genesis.json by calling `initialize()` on all of the predeploys and spitting out a network specific genesis.json

Reply:
[a06cdf2](https://github.com/ethereum-optimism/optimism/pull/10320/commits/a06cdf22a06e1db8529f302a4f1eb87cfd45c850)